frontend/src/pages/MetricCreateEdit.jsx

```diff
@@ -0,0 +1,181 @@
+        getMetricById(metric_id)
+        if(props.mode == 'view'){
+            disable = 'true'
+        }
+    }
+  }, [metric_id])
+
+  return (
+    <Layout>
+        <div className="content mt-4 mb-5" style={{width: '100%'}}>
+            <div className="container">
+                <div className="row">
+                    <div className="col-sm-6 m-auto mb-5">
+                        <h1 className="text-capitalize">{props.mode} Metric</h1>
+                        <form data-mode="create" className="row g-3" id="metric-form"
+                        onKeyDown={(e)=> {return e.key !== 'Enter'}}
+                        onSubmit={(e) => e.preventDefault()}
+                        >
+                            <div className="col-12">
+                                <label htmlFor="metric-name" className="form-label">Metric Name</label>
+                                <input ref={metricNameRef} type="text" className="form-control"
+                                placeholder="" id="metric-name"></input>
+                                <span className="error-msg text-danger d-none">error</span>
+                            </div>
+                            <div className="col-12">
+                                <label htmlFor="metric-type" className="form-label">Metric Type</label>
+                                <select ref={metricTypeRef}
+                                className="form-select" name="" id="metric-type">
+                                    <option value="integer">integer</option>
+                                    <option value="boolean">boolean</option>
+                                </select>
+                                <span className="error-msg text-danger d-none">error</span>
+                            </div>
+                            <div className="col-12">
+                                <TagsBox label="Metric Area" tags={tags} setTags={setTags}/>
+                                <span className="error-msg text-danger d-none">error</span>
+                            </div>
+                            <div className="col-md-12">
+                                <label htmlFor="metric-description" className="form-label">Metric Description </label>
+                                <textarea  ref={metricDescRef} className="form-control" 
+                                placeholder="Write a brief descriptoin here" maxLength="100" 
+                                id="metric-description" style={{height: '100px'}}
+                                onChange={(e)=>setCharLimit(e.target.value.length)}></textarea>
+                                <div className="float-end"><span className="counter">{charLimit}</span>/100</div>
+                                <span className="error-msg text-danger d-none">error</span>
+                            </div>
+                            <div className="col-12">
+                                <button type="submit" className="btn btn-primary col-12" 
```

Comment:
setting the type to 'submit' will submit the form. so instead of suppressing the form submission on the form element and listen to onClick no the button, remove the onClick handler here and set the onSubmit to your function